addons/analytic/migrations/1.2/pre-migrate.py

```diff
@@ -0,0 +1,14 @@
+def migrate(cr, version):
+    cr.execute("SELECT id FROM account_analytic_plan")
+    plan_ids = [r[0] for r in cr.fetchall()]
+    column_names = [f"x_plan{id_}_id" for id_ in plan_ids]
+    cr.execute(
+        """
+        UPDATE ir_model_fields
+           SET on_delete = 'restrict'
+         WHERE model = 'account.analytic.line'
+           AND on_delete = 'set null'
```

Comment:
You also need to change the actual constraint on the SQL table definition.
Have a look at `add_foreign_key`